ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/clientlibs/editorhook/js/replacehook.js

```diff
@@ -0,0 +1,266 @@
+
+        return (
+            sendReplaceParagraph({
+                resourceType: component.getResourceType(),
+                configParams: component.getConfigParams(),
+                extraParams: component.getExtraParams(),
+                templatePath: component.getTemplatePath()
+            }, editable)
+                .done(function () {
+                    window.CQ.FormsCoreComponents.editorhooks._refreshEditable(editable, component);
+                    channel.trigger("cq-persistence-after-replace", args);
+                })
+                .fail(function () {
+                    author.ui.helpers.notify({
+                        content: Granite.I18n.get("Paragraph replace operation failed."),
+                        type: author.ui.helpers.NOTIFICATION_TYPES.ERROR
+                    });
+                })
+        );
+    };
+
+    window.CQ.FormsCoreComponents.editorhooks._refreshEditable = function (editable, component) {
+        editable.refresh().done(function () {
+            var callbackInvoked = false;
+            var addedComponentPath = component.path;
+            if (addedComponentPath) {
+                guideTouchLib.editLayer.Interactions.onOverlayClick({
+                    editable : Granite.author.editables.find(addedComponentPath)[0]
+                });
+                guidelib.author.editConfigListeners.addedComponentPath = undefined;
+            }
+        });
+    };
+
+    /**
+     * Send a request to replace an existing component
+     * @param {String} config.resourceType The resource type of the new paragraph.
+     * @param {Object} [config.configParams] The config parameters to be set upon the new paragraph's creation.
+     * @param {Object} [config.extraParams] The extra parameters (would override other params) to be set upon the new paragraph's creation.
+     * @param  {String} [config.templatePath] The path to the template definition that should be used.
+     * @param {Object} [editable] editable to replace
+     * @return {$.Deferred} A deferred object that will be resolved when the request is completed.
+     */
+    var sendReplaceParagraph = function (config, editable) {
+        return (
+            new author.persistence.PostRequest()
+                .prepareReplaceParagraph(config, editable)
+                .send()
+        );
+    };
+
+    author.persistence.PostRequest.prototype.prepareReplaceParagraph = function (config, editable) {
+        // apply component properties over editable
+        if (config.templatePath) {
+            var comTemplatePath = Granite.HTTP.externalize(config.templatePath),
+                comData = CQ.shared.HTTP.eval(comTemplatePath + ".infinity.json"); // get component default json
+            editableJson = CQ.shared.HTTP.eval(editable.path + ".infinity.json"); // get editable json
+            comGuideNodeClass = comData.guideNodeClass; // guideNodeClass has to be written in editable so preserve it
```

Comment:
We dont have anything called as guideNodeClass. this will not work.
how is this code tested?